3ds Max/Max2Babylon/Exporter/BabylonExporter.Mesh.cs

```diff
@@ -603,29 +607,124 @@ private void ExportUserData(IIGameNode meshNode, BabylonAbstractMesh babylonMesh
+                babylonMesh.visibility = maxNode.MaxNode.GetVisibility(0, Tools.Forever);
+
+                // Material
+                var mtl = maxNode.NodeMaterial;
+                var multiMatsCount = 1;
+
+                // The DirectXShader material is a passthrough to its render material.
+                // The shell material is a passthrough to its baked material.
+                while (mtl != null && (isShellMaterial(mtl) || isDirectXShaderMaterial(mtl)))
+                {
+                    if (isShellMaterial(mtl))
+                    {
+                        // Retrieve the baked material from the shell material.
+                        mtl = GetBakedMaterialFromShellMaterial(mtl);
+                    }
+                    else // isDirectXShaderMaterial(mtl)
+                    {
+                        // Retrieve the render material from the directX shader
+                        mtl = GetRenderMaterialFromDirectXShader(mtl);
+                    }
+                }
+
+                isMaterialDoubleSided = false;
+                if (mtl != null)
+                {
+                    IIGameMaterial unsupportedMaterial = isMaterialSupported(mtl);
+                    if (unsupportedMaterial == null)
+                    {
+                        babylonMesh.materialId = mtl.MaxMaterial.GetGuid().ToString();
+
+                        if (!referencedMaterials.Contains(mtl))
+                        {
+                            referencedMaterials.Add(mtl);
+                        }
+
+                        multiMatsCount = Math.Max(mtl.SubMaterialCount, 1);
+
+                        if (isDoubleSidedMaterial(mtl))
+                        {
+                            isMaterialDoubleSided = true;
+                        }
+                    }
+                    else
+                    {
+                        if (mtl.SubMaterialCount == 0 || mtl == unsupportedMaterial)
+                        {
+                            RaiseWarning("Unsupported material type '" + unsupportedMaterial.MaterialClass + "'. Material is ignored.", 2);
+                        }
+                        else
+                        {
+                            RaiseWarning("Unsupported sub-material type '" + unsupportedMaterial.MaterialClass + "'. Material is ignored.", 2);
+                        }
+                    }
+                }
+                else
+                {
+                    // ensure to clean the material because Clone() did not define any behavior about
+                    // cloning the material id.
+                    babylonMesh.materialId = null;
+                }
+
+                // Export the custom attributes of this mesh
+                babylonMesh.metadata = ExportExtraAttributes(maxNode, babylonScene);
+
+                // Append userData to extras
+                ExportUserData(maxNode, babylonMesh);
```

Comment:
Will this result in us having user data from the original mesh, and the clone on the exported clone?

Reply:
yes, data are duplicated

Reply:
Are we sure thats the intention here? I was under the assumption that user data is instance-specific, I'm not sure that we want to duplicate this data as well.

Reply:
Well, in max, Instance are a "pointer" to a mesh, and then share ALL the values and behavior. Who can clarify the initial intention ?